NuRadioReco/modules/channelGalacticNoiseAdder.py

```diff
@@ -0,0 +1,209 @@
+                ax_2.set_yscale('log')
+                ax_3.grid()
+                ax_3.set_yscale('log')
+                ax_4.grid()
+                ax_4.set_yscale('log')
+                ax_1.set_xlabel('f [MHz]')
+                ax_2.set_xlabel('f [MHz]')
+                ax_3.set_xlabel('f [MHz]')
+                ax_4.set_xlabel('f [MHz]')
+                ax_1.set_ylabel('T [K]')
+                ax_2.set_ylabel('S [W/m²/MHz]')
+                ax_3.set_ylabel('E [V/m]')
+                ax_4.set_ylabel('U [V]')
+                ax_4.plot(channel.get_frequencies() / units.MHz, np.abs(channel.get_frequency_spectrum()), c='C0')
+                ax_4.set_ylim([1.e-8, None])
+                fig1 = plt.figure()
+                ax1 = fig1.add_subplot(211)
+                ax2 = fig1.add_subplot(212)
+                ax1.plot(channel.get_times(), channel.get_trace(), label='original trace')
+                ax2.plot(channel.get_frequencies() / units.MHz, np.abs(channel.get_frequency_spectrum()), label='original spectrum')
+                ax1.grid()
+                ax2.grid()
+            for i_pixel in range(healpy.pixelfunc.nside2npix(self.__n_side)):
+                azimuth = local_coordinates[i_pixel].az.rad
+                zenith = np.pi / 2. - local_coordinates[i_pixel].alt.rad
+                if zenith > 90. * units.deg:
+                    continue
+                temperature_interpolator = scipy.interpolate.interp1d(self.__interpolaiton_frequencies, np.log10(noise_temperatures[:, i_pixel]), kind='quadratic')
+                noise_temperature = np.power(10, temperature_interpolator(freqs[passband_filter]))
+                S = (2. * scipy.constants.Boltzmann * (freqs[passband_filter] / units.Hz)**2 / scipy.constants.c**2 * (noise_temperature / units.kelvin) * solid_angle) * (units.watt / units.m**2 / units.Hz)
+                S[np.isnan(S)] = 0
+                S_per_bin = S * d_f
+                flux_sum += S_per_bin
+                E = np.sqrt(S_per_bin / (units.watt / units.m**2) / (scipy.constants.c * scipy.constants.epsilon_0)) / (d_f) * units.V / units.m
+                if self.__debug:
+                    ax_1.scatter(self.__interpolaiton_frequencies / units.MHz, noise_temperatures[:, i_pixel] / units.kelvin, c='k', alpha=.01)
+                    ax_1.plot(freqs[passband_filter] / units.MHz, noise_temperature, c='k', alpha=.02)
+                    ax_2.plot(freqs[passband_filter] / units.MHz, S_per_bin / d_f / (units.watt / units.m**2 / units.MHz), c='k', alpha=.02)
+                    ax_3.plot(freqs[passband_filter] / units.MHz, E / (units.V / units.m), c='k', alpha=.02)
+
+                noise_spectrum = np.zeros((3, freqs.shape[0]), dtype=np.complex)
+                phases = np.random.uniform(0, 2. * np.pi, len(S))
+                polarizations = np.random.uniform(0, 2. * np.pi, len(S))
+                noise_spectrum[1][passband_filter] = E * np.exp(1j * phases) * np.cos(polarizations)
```

Comment:
I think what is missing here is the application of the Fresnel transmission coefficient.There are actually two things:
1) the signal is attenuated by refracting into the ice
2) the signal direction changes, so the antenna pattern needs to evaluated at a different location. This point is already taken into account by the `get_efield_antenna_factor` function but only if the station type is set to `cosmic_ray`. Is this the case?

Reply:
Fixed